packages/core/src/domain/session/sessionManager.spec.ts

```diff
@@ -614,16 +616,38 @@ describe('startSessionManager', () => {
     })
   })
 
+  describe('session tracking type transition', () => {
+    it('should expire the session when the tracking type transitions to unallowed type', () => {
+      const sessionManager = startSessionManagerWithDefaults({
+        allowedTrackingTransition: FakeTrackingType.OTHER_TRACKING,
+      })
+      sessionManager.updateSession({ [FIRST_PRODUCT_KEY]: FakeTrackingType.DISALLOWED_TRACKING })
+
+      expect(sessionManager.findActiveSession()).not.toBeDefined()
+    })
+    it('should update current entity in history', () => {
+      const sessionManager = startSessionManagerWithDefaults({
+        allowedTrackingTransition: FakeTrackingType.OTHER_TRACKING,
+      })
+      sessionManager.updateSession({ [FIRST_PRODUCT_KEY]: FakeTrackingType.OTHER_TRACKING })
+
+      expect(sessionManager.findActiveSession()!.id).toBeDefined()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏ 
```suggestion
      expectSessionIdToBeDefined()
```